src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeDirectiveFormattingTest.cs

```diff
@@ -940,13 +945,121 @@ public async Task CodeBlock_ObjectCollectionArrayInitializers(bool useSourceText
             Bar = new string[] {
                 ""Hello"",
                 ""There""
+            },
+            Baz = new string[]
+            {
+                ""Hello"",
+                ""There""
             }
         }
     };
 }
 ");
         }
 
+        [Theory]
+        [CombinatorialData]
+        [WorkItem("https://github.com/dotnet/razor-tooling/issues/6092")]
+        public async Task CodeBlock_ArrayInitializers(bool useSourceTextDiffer)
+        {
+            // The C# Formatter doesn't touch these types of initializers, so nor do we. This test
+            // just verifies we don't regress things and start moving code around.
+            await RunFormattingTestAsync(useSourceTextDiffer: useSourceTextDiffer,
+input: @"
+@code {
+    private void M()
+    {
+        var entries = new string[]
+        {
+            ""a"",
+            ""b"",
+            ""c""
+        };
+    }
+}
+",
+expected: @"@code {
+    private void M()
+    {
+        var entries = new string[]
+        {
+            ""a"",
+            ""b"",
+            ""c""
+        };
+    }
+}
+");
+        }
+
+        [Theory]
+        [CombinatorialData]
+        [WorkItem("https://github.com/dotnet/razor-tooling/issues/6092")]
+        public async Task CodeBlock_ObjectInitializers(bool useSourceTextDiffer)
+        {
+            // The C# Formatter doesn't touch these types of initializers, so nor do we. This test
+            // just verifies we don't regress things and start moving code around.
```

Comment:
What if we had the `RunFormattingTestAsync` run formatting a 3rd time on every expected result to ensure things are `stable`? Would that also cover this scenario?

Reply:
Good idea.. I wonder what will break :)

Reply:
Our testing infrastructure is not up to this task :(

_Most_ of the tests passed without issue, but 8% failed, and doing a spot check of those are either for some unknown reason in the formatting tests, or in a way that doesn't repro when doing multiple format operations in real life.

Good idea for the integration tests though